### PR TITLE
refactor queries.get_app_interface_settings() to support returning typed data

### PR DESCRIPTION
**Pull request by jfchevrette**
This refactors `get_app_interface_settings` method to support returning typed data.

The query logic has been extracted into a `query` method that can be re-used everywhere we use a `gql_api` and run a query, returning a dict by default for backward compatibility

Using this new query method `get_app_interface_settings` is changed to return typed data by default and all places where we use the method and expect a dict has been updated with the `typed=False` argument. Old code can be migrated over time to use typed data.

The `query` method uses pydantic's `model.dict()` to export the model to a dictionary, preserving the original graphql query keys using the `by_alias` argument which is originally set by qenerate when generating the models.

Simple tests were also added to test that the returned data type is as expected

Ultimately, this could serve as a pattern for migrating all existing graphql queries to typed data

### CODE REVIEW

**Comment by jfchevrette** [comment]
It just occurred to me that it is probably preferable to invert the condition and make the method return typed data by default. I can easily add `typed=False` to all places where `get_app_interface_settings()` is used, which is in about ~100 places.

Thoughts?

**Comment by jfchevrette** [comment]
As I suggested, I made the method return typed data by default and updated all uses of the method to include a `typed=False` argument.

I also extracted the query logic into a separate `query(...)` method which we should be able to use everywhere else we initialize a `gql_api` client and run a query

**Comment by fishi0x01** [comment]
Sorry for dropping in on this one so late. There was recently a discussion with @geoberle and @steveteahan about how integrate generated classes into our codebase.

This is the [proposal](https://gitlab.cee.redhat.com/service/dev-guidelines/-/merge_requests/134).

The issue we are trying to address is that every query has its own generated data classes. e.g., there might be many different `ApplicationV1` instances scattered across the code base. We thought, that ideally generated classes will not be shared across integrations. Instead, for shared classes, we use `Protocol`. I am wondering if `app_interface_settings` might also be a place which could benefit from Protocol.

**Comment by fishi0x01** [comment]
That being said, I still love the refactor :) :heart: Just wondering about your opinion on the proposal above.

**Comment by jfchevrette** [comment]
Since app_interface_settings is pretty much used throughout all integrations and more, I think it could be prime candidate for this proposal? I also think that app_interface_settings are simple enough that "one query to rule them all" might also be a good option to avoid duplicating slightly different queries all around.

Where we have more complex schemas like `NamespaceV1`, which can appear in multiple different queries, selecting different fields and whatnot, I think is what would benefit the most from this proposal

I'm happy to close this PR for now and wait until the right patterns for generated classes is decided and finalized

File: reconcile/queries.py [hunk 1]
```diff
@@ -31,86 +61,11 @@ def get_secret_reader_settings() -> Optional[Mapping[str, Any]]:
...
-
-def get_app_interface_settings():
-    """Returns App Interface settings"""
-    gqlapi = gql.get_api()
-    settings = gqlapi.query(APP_INTERFACE_SETTINGS_QUERY)["settings"]
-    if settings:
-        # assuming a single settings file for now
-        return settings[0]
-    return None
+def get_app_interface_settings(typed: bool = True):
```
[the remark this answers is not in the document]
Comment (jfchevrette) [reply reconcile/queries.py:64]: I thought of doing this at first but then everywhere else we use `get_app_interface_settings()` we also have to check the type of the returned value before we do either `data["somekey"]` or `data.somekey` as mypy complains that the object either doesn't implement `get()` or doesn't implement `somekey`

I could not find a clean way around this ...

**Pull request closed without merging**